src/server/api/middlewares/bearer-token.ts

```diff
@@ -48,19 +49,12 @@ async function authenticateWithAccessToken(
 
   const accessToken = await findAccessToken(clientId, c);
 
-  if (!accessToken) {
```

Comment:
The early bailout is vulnerable to timing attacks because the attacker can verify whether the ID exists in the database based on the timing.